File: app/src/main/java/app/vodio/com/vodio/fragments/BottomNavFragment.java
package app.vodio.com.vodio.fragments;

import android.content.Context;
import android.net.Uri;
import android.os.Bundle;

import androidx.appcompat.view.menu.MenuItemImpl;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;

import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.tbuonomo.morphbottomnavigation.MorphBottomNavigationView;

import java.util.Collections;

import app.vodio.com.vodio.R;
import app.vodio.com.vodio.activities.HomeActivity;


public class BottomNavFragment extends AbstractFragment {
    private ImageButton recordVodBottom;
    private BottomNavigationView navbar;

    private int[] bottomItemIds = {R.id.bedroom_navbar,R.id.salon_navbar, R.id.profile_navbar,R.id.actu_navbar,R.id.settings_navbar};
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.fragment_bottom_nav, container, false);
        recordVodBottom=v.findViewById(R.id.recordVodFHome);
        navbar = v.findViewById(R.id.bottomnavbar);
        // Inflate the layout for this fragment
        return v;
    }

    @Override
    public void onResume() {
        super.onResume();
        recordVodBottom.setOnClickListener(this);
        navbar.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(MenuItem menuItem) {
                HomeActivity homeActivity = (HomeActivity) parent;
                int position = getPositionByItemId(menuItem.getItemId());
                homeActivity.setPagePosition(position);
                return false;
            }
        });
    }

    @Override
    public void onClick(View view) {
        onItemSelected(view.getId());
    }

    public void setSelected(int position){
        navbar.setSelectedItemId(getItemIdByPosition(position));
    }

    private int getItemIdByPosition(int position){
        return bottomItemIds[position];
    }
    private int getPositionByItemId(int itemId){
        for(int i = 0 ; i < bottomItemIds.length; i++){
            if(bottomItemIds[i] == itemId){
                return i;
            }
        }
        return 0;
    }
}

File: app/src/main/java/app/vodio/com/vodio/fragments/BottomNavFragment.kt
package app.vodio.com.vodio.fragments

import android.os.Bundle
import android.util.SparseArray

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment

import app.vodio.com.vodio.R
import app.vodio.com.vodio.activities.HomePagerActivity
import app.vodio.com.vodio.fragments.utils.FragmentIcon
import com.google.android.material.bottomnavigation.BottomNavigationView
import kotlinx.android.synthetic.main.fragment_bottom_nav.*


class BottomNavFragment : AbstractFragment() {
    private val mapFragmentMenu = SparseArray<FragmentIcon>()
    var fragments : List<Fragment> = listOf(ChambreFragment(), SalonFragment(), ProfileFragment(), ActuFragment(), SearchFragment())

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_bottom_nav, container, false)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        for(fragment in fragments){
            val fragmentIcon = fragment as FragmentIcon
            val id = if(fragmentIcon.getIconId() != null) fragmentIcon.getIconId() else R.drawable.ic_settings_icon_white
            mapFragmentMenu.put(id!!, fragment)
        }
    }

    override fun onResume() {
        super.onResume()
        recordVodFHome.setOnClickListener(this)
        bottomnavbar.setOnNavigationItemSelectedListener { menuItem ->
            val homeActivity = parent as BottomNavigationView.OnNavigationItemSelectedListener?
            homeActivity!!.onNavigationItemSelected(menuItem)

            //val position = getPositionByItemId(menuItem.itemId)
            //setSelected(position)
            //homeActivity!!.setPagePosition(position)
            false
        }
    }

    override fun onClick(view: View) {
        onItemSelected(view.id)
    }

    fun setSelected(position: Int) {
        bottomnavbar.selectedItemId = getItemIdByPosition(position)
    }

    private fun getItemIdByPosition(position: Int): Int {
        return (fragments[position] as FragmentIcon).getIconId()!!
    }

    fun getPositionByItemId(itemId: Int): Int {
        for (i in fragments.indices) {
            if ((fragments.get(i)!! as FragmentIcon).getIconId()  == itemId) {
                return i
            }
        }
        return -1
    }

    override fun getIconId(): Int? {
        return null
    }
}
